docs/writing_to_promscale.md

```diff
@@ -0,0 +1,173 @@
+
+## JSON streaming format
+
+This format was introduced in Promscale to enable easier usage of the endpoint when ingesting metric data from 3rd party tools. It is not part of the `remote_write` specification for Prometheus. It is slightly less efficient to use this format than the Protobuf format. 
+
+
+This format is a JSON stream of objects with two fields:
+* `labels` JSON object in which all the labels are represented as fields
+* `samples` JSON array which contains arrays of timestamp-value sets
+
+Here is a short example of what a simple payload would look like:
+
+```
+{
+    "labels":{"__name__": "cpu_usage", "namespace":"dev", "node": "brain"},
+    "samples":[
+        [1577836800000, 100],
+        [1577836801000, 99],
+        [1577836802000, 98],
+        ...
+    ]
+}
+{
+    "labels":{"__name__": "cpu_usage", "namespace":"prod", "node": "brain"},
+    "samples":[
+        [1577836800000, 98],
+        [1577836801000, 99],
+        [1577836802000, 100],
+        ...
+    ]
+}
+...
+```
+
+As you can see, the labels object is pretty self explanatory. Fields represent the label name and value is the label value. Samples array consists of arrays with some strict rules:
+* There can be only two values in the array
+* First value is a UNIX timestamp in milliseconds which has to be an integer value
+* Second value is a floating point number which represents the corresponding value of the metric.
+
+
+In order to send a request to Promscale, you would need to send an HTTP POST request with the request body set to the JSON payload and set the required header values:
+* `Content-Type` header should be set to `application/json`
+* `Content-Encoding` header can optionally be set to `snappy` if you are using snappy compression on the JSON payload
+
+
+Here is an example of a request sent using `curl` tool:
+
+```
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"labels":{"__name__":"foo"},"samples":[[1577836800000, 100]]}' \
+http://localhost:3000/api/login
+```
+
+## Protobuf format
+
+This format is a little more complex and involved to use than the JSON format but generally is a bit more performant because the payload is smaller in size. It is recommended that you use this format if you are building a system which would ingest a lot of data continuously.
+
+In order to use the endpoint, there are a few steps that need to be done:
+* Fetch the protocol buffer definition files [here](https://github.com/prometheus/prometheus/blob/master/prompb/)
+* [Compile them](https://developers.google.com/protocol-buffers/docs/tutorials) in into structures of the programming language you intend to use
+* Finally, use those structures to construct requests which you can then send to the Promscale write endpoint
+Next section will show a simple example of how to make a request to Promscale using the Go programming language.
+
+
+## Protobuf write request example in Go
+
```

Comment:
Not sure if it is done or not, but having another example that explains JSON way too would be very helpful for the users.

Reply:
I think the curl serves as that example